ush/python/pygfs/task/marine_bmat.py

```diff
@@ -0,0 +1,364 @@
+
+        # Copy MOM6 restart
+        # TODO: check if we can combine this with the step above
+        logger.info(f"Linking MOM6 restart to ocn.bkg.nc")
+        rst_list = []
+        rst_list.append([os.path.join(self.task_config.DATA, "ocn.bkg.nc"),
+                         os.path.join(self.task_config.DATA, "INPUT", "MOM.res.nc")])
+        rst_list.append([os.path.join(self.task_config.DATA, "ice.bkg.nc"),
+                         os.path.join(self.task_config.DATA, "INPUT", "cice.res.nc")])
+        FileHandler({'copy': rst_list}).sync()
+
+        # stage the grid generation yaml
+        FileHandler({'copy': [[self.task_config.GRID_GEN_YAML,
+                               os.path.join(self.task_config.DATA, 'gridgen.yaml')]]}).sync()
+
+        # generate the variance partitioning YAML file
+        logger.debug("Generate variance partitioning YAML file")
+        diagb_config = parse_j2yaml(path=os.path.join(self.task_config.BERROR_YAML_DIR, 'soca_diagb.yaml.j2'),
+                                    data=self.task_config)
+        diagb_config.save(os.path.join(self.task_config.DATA, 'soca_diagb.yaml'))
+
+        # generate the vertical decorrelation scale YAML file
+        logger.debug("Generate the vertical correlation scale YAML file")
+        vtscales_config = parse_j2yaml(path=os.path.join(self.task_config.BERROR_YAML_DIR, 'soca_vtscales.yaml.j2'),
+                                       data=self.task_config)
+        vtscales_config.save(os.path.join(self.task_config.DATA, 'soca_vtscales.yaml'))
+
+        # generate vertical diffusion scale YAML file
+        logger.debug("Generate vertical diffusion YAML file")
+        diffvz_config = parse_j2yaml(path=os.path.join(self.task_config.BERROR_YAML_DIR, 'soca_parameters_diffusion_vt.yaml.j2'),
+                                     data=self.task_config)
+        diffvz_config.save(os.path.join(self.task_config.DATA, 'soca_parameters_diffusion_vt.yaml'))
+
+        # generate the horizontal diffusion YAML files
+        if True:  # TODO(G): Missing logic to skip this section
+            # stage the correlation scale configuration
+            logger.debug("Generate correlation scale YAML file")
+            FileHandler({'copy': [[os.path.join(self.task_config.BERROR_YAML_DIR, 'soca_setcorscales.yaml'),
+                                   os.path.join(self.task_config.DATA, 'soca_setcorscales.yaml')]]}).sync()
+
+            # generate horizontal diffusion scale YAML file
+            logger.debug("Generate horizontal diffusion scale YAML file")
+            diffhz_config = parse_j2yaml(path=os.path.join(self.task_config.BERROR_YAML_DIR, 'soca_parameters_diffusion_hz.yaml.j2'),
+                                         data=self.task_config)
+            diffhz_config.save(os.path.join(self.task_config.DATA, 'soca_parameters_diffusion_hz.yaml'))
+
+        # hybrid EnVAR case
+        if self.task_config.DOHYBVAR == "YES" or self.task_config.NMEM_ENS > 3:
```

Comment:
Members greater than 3, why can't three members work?

Reply:
it should, this is dating back to what we were doing with bump. Changed so that 3 should work.